pulsar/internal/commands.go

```diff
@@ -119,6 +121,20 @@ func (r *MessageReader) ReadMessageMetadata() (*pb.MessageMetadata, error) {
 	return &meta, nil
 }
 
+func (r *MessageReader) ReadBrokerMetadata() (*pb.BrokerEntryMetadata, error) {
+	magicNumber := binary.BigEndian.Uint16(r.buffer.Get(r.buffer.ReaderIndex(), 2))
+	if magicNumber != magicBrokerEntryMetadata {
+		return nil, nil
+	}
```

Comment:
Why is the error object returned as nil here? If we don't need this error field, maybe we can cancel it in function definition? Or we can use `errors.new("xxx")` to customize the error message

Reply:
It's not an error, it means broker not enable `brokermetadata` feature